tests/console/validate_repos.pm

```diff
@@ -19,16 +19,35 @@ sub run {
 
     select_console 'root-console';
 
-    foreach my $repo (@{$test_data->{repos}}) {
-        my $filter = $repo->{filter} ? $repo->{$repo->{filter}} : undef;
-        validate_repo_properties({
-                Filter => $filter,
-                Alias => $repo->{alias},
-                Name => $repo->{name},
-                URI => $repo->{uri},
-                Enabled => $repo->{enabled},
-                Autorefresh => $repo->{autorefresh}
-        });
+    if (check_var("RMT_SERVER", "default") || !get_var("RMT_SERVER")) {
```

Comment:
for sure we need to separate this in two test modules, we try to avoid this `if/else` at the end someone else would add yet another complicated condition to this line. I also not familiar with value 'default', not sure what you wanted to express there :)
do you know that you can use in test data expansion as well?
%ARCH% would it help to have just one module?

Reply:
I too find the creation of a new validation module a better idea. A few lines of similar code are better than hard to read code. Will move it to each own module.

Reply:
all that you use here seems to be just openQA settings `URI => join('', /$rmt_uri/, /$repo->{uri}/, /$ver/, /$sep/, /$arch/, /$repo->{suffix}/, /$ver/)` so there is not need to have this complex concatenation, we can add interpolation with %var% in `test_data` in yaml, then one test module is enough for the uri, right?

Reply:
I did not know that, in that case yes it should be enough. Let me rewrite the test with no new module and see if it works!

Reply:
@jknphy I did follow your advise, however a change is needed in the `validate_repos` string for this to work. I suspect the existance of `/?` in the URI string causes this. Please see the change in the `validate_repos` pm, it was the only way I could get this to work.